homeassistant/components/dsmr/config_flow.py

```diff
@@ -0,0 +1,27 @@
+"""Config flow for DSMR integration."""
+import logging
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DSMRFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for DSMR."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_import(self, import_config=None):
+        """Handle the initial step."""
+        if CONF_HOST in import_config:
+            name = f"{import_config[CONF_HOST]}:{import_config[CONF_PORT]}"
+        else:
+            name = import_config[CONF_PORT]
+
+        await self.async_set_unique_id(name)
```

Comment:
im not sure this HOST:PORT combo is unique enough to HA standards. youll likely need to do an alternativr check like checking againt host and port:

https://github.com/home-assistant/core/blob/ed104d19272e31bd323f719fbdc7f388ccafc85f/homeassistant/components/harmony/config_flow.py#L89

Reply:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/transmission/config_flow.py#L58

Reply:
It worried me a bit as well, but then I saw integrations using username or latitude/longitude as unique id:
https://github.com/home-assistant/core/blob/b71cbd2033097b3ebeeecb6ebe6f2c2b2d70ad65/homeassistant/components/airly/config_flow.py#L36
I assumed it was maybe unique to the integration. If I look into `_abort_if_unique_id_configured` it seems it queries only entries for the integration so uniqueness may not be an issue.

I'd like to stick to the logic where we use `self._abort_if_unique_id_configured(import_config)` because it both updates the data and reloads the integration. Updating data is needed as long as there is no UI alternative in place to update data. The example given use an alternative abort method (`async_abort`).

What'd you think?

Reply:
so in those cases, only 1 username could exist within the service and coordinates are unique. the issue is that a device can move between IPs or networks and it could be on different ports in such situations. this is why historically host/port have been considered not unique

Reply:
I added a custom version above but the general flow of things is that once you import you would then check use that as the source of truth and not import anything further esp for integrations that only have singleinstance in config. there are some exceptions with like homekit where one may have more advanced configurations. (https://github.com/home-assistant/core/blob/dev/homeassistant/components/homekit/__init__.py#L171)

Reply:
> Updating data is needed as long as there is no UI alternative in place to update data.

you can trigger a config flow to get new infomation, some integrations do it for re-auth, i suspect the issue becomes with not knowing when to trigger such (ie temp failures)? You can also add options to your config entries for somethings that may need tweaked more often. generally though if your device changes enough, you would just readd via a manual user flow.

Reply:
The reason to update data and use yaml as source of truth and update data is that this is only a first step. I'm planning a config and option flow PR and once that is in, import should be a one time thing.

I think we could do without as well, but in that case when a user updates yaml (which is the only way to change something) it would first be required to remove the integration on the integrations page before restarting ha. That would however be a very temporary thing because I have code for config and options flow ready.